eventsources/sources/awssqs/start.go

```diff
@@ -112,6 +111,7 @@ func (el *EventListener) StartListening(ctx context.Context, dispatch func([]byt
 		messages, err := fetchMessages(ctx, sqsClient, *queueURL.QueueUrl, 10, sqsEventSource.WaitTimeSeconds)
 		if err != nil {
 			log.Errorw("failed to get messages from SQS", zap.Error(err))
+			_, sqsClient, err = refreshSessionAndSqsClientIfCredentialsExpired(err, el, log)
```

Comment:
What would happen in case `refreshSessionAndSqsClientIfCredentialsExpired` fails? It would return `nil` as the second return value which means `sqsClient` would become nil. This would crash the process because of a nil pointer dereference the next time it tries to call `fetchMessages`..

Reply:
Agreed. Added a nil check before reassigning sqsClient